source/slang/ir-specialize.cpp

```diff
@@ -1145,6 +1153,145 @@ struct SpecializationContext
         }
     }
 
+    IRType* tryGetPointedToType(
+        IRBuilder*  builder,
+        IRType*     type)
+    {
+        if( auto ptrType = as<IRPtrTypeBase>(type) )
+        {
+            return ptrType->getValueType();
+        }
+        else if( auto ptrLikeType = as<IRPointerLikeType>(type) )
+        {
+            return ptrLikeType->getElementType();
+        }
+        else if( auto bindExistentials = as<IRBindExistentialsType>(type) )
+        {
+            auto baseType = bindExistentials->getBaseType();
+            if( auto baseElementType = tryGetPointedToType(builder, baseType) )
+            {
+                UInt existentialArgCount = bindExistentials->getExistentialArgCount();
+                List<IRInst*> existentialArgs;
```

Comment:
Initialize with size perhaps and/or perhaps having a function that just did this setup like 'initIRInstArray(const IRUse* uses, int numUses, List<IRInst>& outArray); as I see the exact same thing later on.

Reply:
Initializing an "empty" instruction with a known number of operands isn't a bad idea.